miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/FileWriter.kt

```diff
@@ -23,14 +24,19 @@ internal fun ZipInputStream.decompress(zipPath: String) = use { input ->
 
     while (input.nextEntry.also { entry = it } != null) {
         val newFile = File(outputFilePath.parentFile, entry!!.name)
-        newFile.parentFile?.mkdirs()
-        if (entry!!.isDirectory)
-            newFile.mkdir()
-        else
-            writeFile(newFile, input)
+        if (newFile.canonicalPath.contains(BuildConfig.LIBRARY_PACKAGE_NAME))
```

Comment:
Yeah, I'm not sure if it's enough to check only `BuildConfig.LIBRARY_PACKAGE_NAME`, maybe it's more safer to give more data e.g. `outputFilePath.parentFile?.canonicalPath.toString()`

Reply:
Also, how about we could throw an exception while there is any vulnerability found, I had an idea e.g.

```
private fun checkZipVulnerability(newFile: File, outputDir: String) {
    if (!newFile.canonicalPath.startsWith(outputDir))
        throw MiniAppSdkException("A vulnerability has been found while zip path traversal with" + newFile.canonicalPath)
}

// Then we can use it as following:

    ...
    ...

    while (input.nextEntry.also { entry = it } != null) {
        ...
        try {
            checkZipVulnerability(newFile, outputFilePath.parentFile?.canonicalPath.toString())
        } catch (e: MiniAppSdkException) {
            e.printStackTrace()
            return@use
        }
        createData(newFile, entry!!, input)
    }
...
```

Reply:
what is `outputDir`?

Reply:
I have passed `outputFilePath.parentFile?.canonicalPath.toString()` as `outputDir`.
Basically, it's the `DIR` which was suggested in https://support.google.com/faqs/answer/9294009
<img width="339" alt="Screen Shot 2021-05-17 at 15 06 34" src="https://user-images.githubusercontent.com/66930373/118439743-8a7a3280-b721-11eb-9729-49c8b7f489b5.png">
We can use canonical paths to verify this issue.

Reply:
ah the `newFile` is generated from the `outputDir` path so that check does not help. My check also doesn't help much, the purpose of this pr is just to remove the meaningless warning.

Reply:
@minh-rakuten But the value of `BuildConfig.LIBRARY_PACKAGE_NAME` is `com.rakuten.tech.mobile.miniapp`. So I think this only works in our Demo App because the package name of our demo app is `com.rakuten.tech.mobile.miniapp.testapp`.

If this ran in another App with the package name `com.example` then the output directory would be  `/data/data/com.example/files/miniapp`, so it looks like it would fail to extract the Zip.

I guess you could use `zipPath` and parse out the filename so you can have just the base directory?

Reply:
updated to check start with canonical base path.